core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJVersion.scala

```diff
@@ -3,7 +3,13 @@ package org.virtuslab.ideprobe.dependencies
 import pureconfig.ConfigConvert
 import pureconfig.generic.semiauto.deriveConvert
 
-final case class IntelliJVersion(build: String, release: Option[String]) {
+import org.virtuslab.ideprobe.Config
+import org.virtuslab.ideprobe.IntelliJFixture
+import org.virtuslab.ideprobe.config.IntellijConfig
+
+// the `ext` field (file extension) must be an Option[String] and not a String for the ExistingIntelliJ usage
+// inside the `org.virtuslab.ideprobe.dependencies.IntelliJVersionResolver` object
+final case class IntelliJVersion(build: String, release: Option[String], ext: Option[String]) {
```

Comment:
Maybe it doesn't need to be optional and could be just ignored by existing IJ?

Reply:
Hm 🤔 What exactly do you have in mind? I don't understand to be honest. The thing is - `IntelliJVersionResolver.version` - used only for `ExistingIntelliJ` - returns an instance of `IntelliJVersion`. And after my changes - `IntelliJVersion` will have the additional `ext` field. But this in fact is valid and makes sense only for using `IntellijConfig.Default` - but makes no sense for `IntellijConfig.Existing`.

What do you suggest?

Reply:
I understand this. This is a bit ugly though to call .get because it is valid in specific context. I was considering using String rather than Option[String], and just put a special value for existing case or ignore it in that context.

Another valid approach would be to store "ext" elsewhere or have some ADT, so that version for downloadable IJ has this field, but version for existing IJ does not have this field. It might be complicated to integrate it with rest of the code nicely though.

Reply:
Changed ext field from Option[String] to String